src/components/Segmented/Segmented.tsx

```diff
@@ -0,0 +1,143 @@
+   * @param {Option} option - The newly selected option.
+   */
+  onChange?: (option: Option) => void
+
+ /**
+   * An array of segmented options to be displayed. Either one of the following:
+   *
+   * - string
+   * - object: {
+   *
+   *     icon?: ReactNode, The icon of the option
+   *
+   *     label: ReactNode, The selectable text of the option
+   *
+   *     isDisabled?: boolean, Whether the option is clickable
+   *
+   *     key: string, The value associated with the option
+   *
+   *   }
+   */
+  options: Option[]
+
+  /**
+   * The alignment of the options within the segmented control. Either one of the following:
+   *
+   * - horizontal: icons and labels of every option are aligned horizontally
+   * - vertical: icons and labels of every option are aligned vertically
+   */
+  optionsAlignment?: OptionsAlignments
+
+ /**
+   * The current selected value. Either one of the following:
+   *
+   * - string: the key of the selected option
+   * - number: the position of the selected option in the array
+   */
+  value?: string | number
+}
+
+/**
+ * UI component for displaying selectable segmented options
+ *
+ * @returns {Segmented} Segmented component
+ */
+export const Segmented = ({
+  defaultValue,
+  isDisabled,
+  onChange,
+  options,
+  optionsAlignment,
+  value,
+}: SegmentedProps): ReactElement => {
+  const [selectedValue, setSelectedValue] = useState(resolveKey(options, defaultValue))
+
+  const handleOptionClick = (option: Option): void => {
+    if (!isDisabledOption(option, isDisabled)) {
+      setSelectedValue(getOptionKey(option))
+      if (onChange) {
+        onChange(option)
+      }
+    }
+  }
+
+  return (
+    <ul
+      className={classnames([
+        segmented,
+        isDisabled && disabled,
+        optionsAlignment === Vertical && vertical,
+      ])}
+    >
+      {
+        options.map((option) => {
+          const currentKey = resolveKey(options, value) ?? selectedValue
+          const key = getOptionKey(option)
+
+          const selectedOption = key === currentKey
+          const verticalOption = isVerticalOption(option, optionsAlignment)
+          const disabledOption = isDisabledOption(option, isDisabled)
+
+          return (
+            <li
+              aria-checked={Boolean(selectedOption)}
+              aria-disabled={Boolean(disabledOption)}
+              aria-label={key}
+              className={classnames([
+                segmentedOption,
+                selectedOption && selected,
+                verticalOption && vertical,
+                disabledOption && disabled,
+              ])}
+              key={key}
+              onClick={() => handleOptionClick(option)}
+            >
+              {
+                isString(option)
+                  ? option
+                  : (
+                    <>
+                      {option.icon}
+                      {option.label}
+                    </>
+                  )
+              }
```

Comment:
these ternaries made like this don't add up to me. We use the default of eslint. https://eslint.org/docs/latest/rules/multiline-ternary

```suggestion
              {isString(option) ? option : <>{option.icon}{option.label}<>}
```

Reply:
OK, you cannot use the "one-line" proposal, but I use the interrogative and exclamatory sentences at the end of the line and not at the start. Let's align ourselves with the default